var/spack/repos/builtin/packages/gdal/package.py

```diff
@@ -94,9 +94,9 @@ class Gdal(AutotoolsPackage):
 
     # Required dependencies
     depends_on('libtiff@3.6.0:')  # 3.9.0+ needed to pass testsuite
-    depends_on('libgeotiff@1.2.1:')
-    depends_on('libgeotiff@:1.4', when='@:2.4')
-    depends_on('libgeotiff@1.5:', when='@3:')
+    depends_on('libgeotiff@1.2.1:1.4', when='@:2.4.0')
+    depends_on('libgeotiff@1.2.1:1.5', when='@2.4.1:')
+    depends_on('libgeotiff@1.5:', when='@3.0.0:')
```

Comment:
Now you're saying that version 3 can only be built with 1.5. I think what you want is:
```python
depends_on('libgeotiff@1.2.1:')
depends_on('libgeotiff@:1.4', when='@:2.4.0')
depends_on('libgeotiff@:1.5', when='@:2')
depends_on('libgeotiff@1.5:', when='@3:')
```

Reply:
I don't like the hanging `depends_on('libgeotiff@1.2.1:')` - in similar cases I have encountered issues with concretizer ignoring later depends_on clauses. But could be wrong.

Reply:
If you encounter that problem again, open an issue with instructions for how to reproduce it. The concretizer is far from perfect, but @tgamblin is working on fixing most of the issues.